examples/array-datatype/AppManifest.json

```diff
@@ -1,18 +1,32 @@
-[
-    {
-        "name": "array-datatype",
-        "vehicleModel": {
-            "src": "https://github.com/COVESA/vehicle_signal_specification/releases/download/v3.0/vss_rel_3.0.json",
-            "datapoints": [
-                {
-                    "path": "Vehicle.Powertrain.FuelSystem.SupportedFuel",
-                    "required": "true",
-                    "access": "read"
+{
+    "manifestVersion": "v3",
+    "name": "array-datatype",
+    "interfaces": [
+        {
+            "type": "vehicle-signal-interface",
+            "config": {
+                "src": "https://github.com/COVESA/vehicle_signal_specification/releases/download/v3.0/vss_rel_3.0.json",
+                "datapoints": {
+                    "required": [
+                        {
+                            "path": "Vehicle.Powertrain.FuelSystem.SupportedFuel",
+                            "optional": "false",
+                            "access": "read"
+                        }
+                    ]
                 }
-            ]
+            }
         },
-        "runtime": [
-            "grpc://sdv.databroker.v1.Broker/Subscribe"
-        ]
-    }
-]
+        {
+            "type": "grpc-interface",
+            "config": {
+                "src": "https://github.com/eclipse/kuksa.val/blob/master/kuksa_databroker/proto/sdv/databroker/v1/broker.proto",
```

Comment:
In case that we keep the databroker interface "referencing" here (see my other finding), shouldn't we specify a fixed version of the proto source (instead of "master")?

Reply:
Master just refers to the repository. I expect the v1 in "proto/sdv/databroker/v1/broker.proto" to be updated if something in the interface changes

Reply:
Yes and no: "v1" is just a major version, whose purpose is to allow to maintain multiple major versions in parallel.
A version "v2" is added once some incompatible change is done in the interface. As long as there are only compatible changes the interface is still maintained as "v1". (There are no minor or patch versions at this level.)
Function-wise this should not harm, as the app still uses the "old" interface (and exactly specifies the methods used). But the advantage of pinning the exact version would be reproducibility. Maybe this is more relevant for C++, because if a new compatible version is released on master ("under the hood" of v1), this will end up in re-compilation.